hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/HoodieSparkSqlWriterSuite.scala

```diff
@@ -336,7 +337,42 @@ class HoodieSparkSqlWriterSuite extends FunSuite with Matchers {
     }
   }
 
-  test("test bulk insert dataset with datasource impl multiple rounds") {
+  test("test read of a table with one failed write") {
```

Comment:
don't think we need a functional test for this. we should be cognizant about over-testing (causing long testing time and maintenance effort). The simple logic flow can be properly covered by a UT on `createRelation()` to assert an exception?